xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/MacroBlock.java

```diff
@@ -52,16 +52,6 @@
      */
     private boolean inline;
 
-    /**
-     * @param id the id of the macro
-     * @param parameters the parameters of the macro
-     * @param isInline indicate if the macro is located in a inline content (like paragraph, etc.)
-     */
-    public MacroBlock(String id, Map<String, String> parameters, boolean isInline)
```

Comment:
You don't explain what it's duplicating and deleting a public API is definitely a major breakage.

You should not blindly apply what you think sonar indicate and think about the consequences. As indicated in the GCI task, sometimes Sonar is wrong and you should ask on the chat when you are not sure.